test/bokken/pairings_test.exs

```diff
@@ -0,0 +1,94 @@
+defmodule Bokken.PairingsTest do
+  @moduledoc false
+  use Bokken.DataCase
+
+  import Bokken.Factory
+
+  describe "pairings" do
+    alias Bokken.Pairings
+
+    defp forget(struct, field, cardinality \\ :one) do
+      %{
+        struct
+        | field => %Ecto.Association.NotLoaded{
+            __field__: field,
+            __owner__: struct.__struct__,
+            __cardinality__: cardinality
+          }
+      }
+    end
+
+    test "get_available_mentors/1 returns all available mentors for an event" do
+      event = insert(:event)
+      skill = insert(:skill)
+
+      mentor =
+        insert(:mentor, %{skills: [skill]})
+        |> forget(:user)
+
+      attrs = %{event: event, is_available: true, mentor: mentor}
+
+      insert(:availability, attrs)
+
+      result = Pairings.get_available_mentors(event.id)
+
+      assert result == [mentor]
+    end
+
+    test "get_available_ninjas/1 returns all available ninjas for an event" do
+      event = insert(:event)
+      skill = insert(:skill)
+
+      ninja =
+        insert(:ninja, %{skills: [skill]})
+        |> forget(:guardian)
+        |> forget(:user)
+
+      attrs = %{event: event, accepted: true, ninja: ninja}
+
+      insert(:enrollment, attrs)
+
+      result = Pairings.get_available_ninjas(event.id)
+
+      assert result == [ninja]
+    end
+
+    test "create_pairings/1 returns a list of lectures for each pairing" do
```

Comment:
Given the algorithmic complexity of this function, one test is not enough to determine its correctness. You should move this to a separate ```describe``` block, where you have more tests with different nuances (like having a ninja which has no common skill to anyone else, different number of ninja / mentors, .etc). Simply put, more tests and more diversity of input cases